adapters/pangle/pangle.go

```diff
@@ -78,23 +85,31 @@ func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapte
 			errs = append(errs, fmt.Errorf("failed unmarshalling imp ext (err)%s", err.Error()))
 			continue
 		}
+		// get token & networkIDs
+		var bidderImpExt openrtb_ext.ImpExtPangle
+		if err := json.Unmarshal(impExt.Bidder, &bidderImpExt); err != nil {
+			errs = append(errs, fmt.Errorf("failed unmarshalling bidder imp ext (err)%s", err.Error()))
+			continue
+		}
 		// detect and fill adtype
-		if adType := getAdType(imp, &impExt); adType == -1 {
+		adType := getAdType(imp, &impExt)
+		if adType == -1 {
 			errs = append(errs, &errortypes.BadInput{Message: "not a supported adtype"})
 			continue
-		} else {
-			impExt.AdType = adType
-			if newImpExt, err := json.Marshal(impExt); err == nil {
-				imp.Ext = newImpExt
-			} else {
-				errs = append(errs, fmt.Errorf("failed re-marshalling imp ext with adtype"))
-				continue
+		}
+		// remarshal imp.ext
+		impExt.AdType = adType
+		impExt.IsPrebid = true
+		if len(bidderImpExt.AppID) > 0 || len(bidderImpExt.PlacementID) > 0 {
```

Comment:
Question: do you expect to have one of AppId and PlacementId or both of them? In description you mentioned `add the optional appid and placementid (which when used, must be used in conjunction)`

Reply:
yeah this is a negligence. I have modified this part. Please take a look at it again. btw, isn't there a bidder param check before going into this adapter? can I assume that it's always both or none here at this part of codes?

Reply:
>  btw, isn't there a bidder param check before going into this adapter? can I assume that it's always both or none here at this part of codes?

Yes. The bidder param check isn't part of the json tests, but is guaranteed to run by PBS-Core. You may assume the check passed in your bidder code.